sandbox/libs/authn/src/main/java/org/opensearch/authn/PermissionFactory.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.authn;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class is used to create Permission instances. The factory can create both standard Permissions which have specific
+ * formatting requirements and legacy permissions which are not checked for validity on creation.
+ */
+public class PermissionFactory {
+
+    public final static String[] INVALID_CHARACTERS = new String[] { ":", "" }; // This is a placeholder for what may want to be banned
+
+    // A placeholder for the different resources which a permission may grant a permission based on
+    public final static String[] QUALIFIED_PERMISSION_TYPES = new String[] { "cluster", "indices", "plugin", "extension" };
+
+    /**
+     * This function creates a standard permission instance. It includes checking that the permission that is being created
+     * is properly formatted.
+     */
+    public Permission createPermission(String permissionString) {
+
+        Permission newPermission = new Permission(permissionString);
+        permissionIsValidFormat(newPermission);
```

Comment:
I had that originally, but we want to make it so that one version does not get checked. I also learned that we want to avoid any of the "side-effects" we can which includes the check being in the constructor.